File: nvxapp.server.service/ClientServer-Service/Account/Models/FinancialAdvisorModel.cs
using nvxapp.server.data.Entities.Public;
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static nvxapp.server.data.Entities.AspNetUsersDataUtil;

namespace nvxapp.server.service.ClientServer_Service.Account.Models
{
    public class FinancialAdvisorModel
    {
        public int IdFinancialAdvisor { get; set; }

        public string? IdAspNetUsers { get; set; }

        public string? Descrizione { get; set; }

    }
    public class FinancialAdvisorListInModel
    {
    }
    public class FinancialAdvisorListOutModel : ModelResult
    {
        public List<FinancialAdvisorModel> FinancialAdvisorList { get; set; }

        public FinancialAdvisorListOutModel()
        {
            FinancialAdvisorList = new List<FinancialAdvisorModel>();
        }
    }




    public class FinancialAdvisorEditModel
    {
        public int IdFinancialAdvisor { get; set; }
        public string IdAspNetUsers { get; set; } = string.Empty;
        public string Descrizione { get; set; } = string.Empty;
        public Boolean MainUser { get; set; }
    }
    public class FinancialAdvisorGetInModel
    {
        public int Id { get; set; } = 0;
    }
    public class FinancialAdvisorGetOutModel : ModelResult
    {
        public FinancialAdvisorEditModel FinancialAdvisorEdit { get; set; } = new FinancialAdvisorEditModel();
    }
    public class FinancialAdvisorPutInModel : ModelResult
    {
        public FinancialAdvisorEditModel FinancialAdvisorEdit { get; set; } = new FinancialAdvisorEditModel();
    }
    public class FinancialAdvisorPutOutModel : ModelResult
    {
        public FinancialAdvisorEditModel FinancialAdvisorEdit { get; set; } = new FinancialAdvisorEditModel();
    }


}

File: nvxapp.client/src/nvx/ClientServer-Service/Account/Models/financial-advisor-model.ts
import { ModelResult } from "../../ModelsBase/model-result";
import { RolesModel } from "./user-roles-model";


export class FinancialAdvisorListModel {
  constructor(
    public idFinancialAdvisor: number = 0,
    public descrizione: string | null = "",
    public idAspNetUsers: string = "",
    public mainUser: boolean = false
  ) { }
}
export class FinancialAdvisorListInModel {
  
}
export class FinancialAdvisorListOutModel extends ModelResult {

  public financialAdvisorList: FinancialAdvisorListModel[] ;

}


export class FinancialAdvisorEditModel {
  constructor(
    public idFinancialAdvisor: number = 0,
    public descrizione: string | null = "",
    public idAspNetUsers: string = "",
    public mainUser: boolean = false,
    public mail: string | null = null,
    public pw: string | null = null

  ) { }
}

export class FinancialAdvisorGetInModel {
  public id: number;
}
export class FinancialAdvisorGetOutModel extends ModelResult {

  public financialAdvisorEdit: FinancialAdvisorEditModel;

}
export class FinancialAdvisorPutInModel  {

  public financialAdvisorEdit: FinancialAdvisorEditModel;

}
export class FinancialAdvisorPutOutModel extends ModelResult {

  public financialAdvisorEdit: FinancialAdvisorEditModel;

}



